Spurning 3

thrainn25 opened the pull request:
Megið endilega fara yfir kóðann sem og útskýringarnar af honum. 

Ef eitthvað er óljóst endilega commenta og ég svara því/laga kóðann.

thrainn25 committed:
Spurning 3

thrainn25 committed:
Endilega farið yfir kóðann sem og rökstuðninginn

thrainn25 committed:
Create Rokstudningur.md

gigjamarin commented on code/Rokstudningur.markdown:
```diff
@@ -0,0 +1,13 @@
+**Reglulega segðin**
+([^,]+)\s([^,]+),\s([^,]+),\s([^,]+),\s(.+)
+
+**tökuhópurinn ([^,]+) sem kemur fram fjórum sinnum**
```
Veit ekki alveg hvað tökuhópur er, spurning að nota annað orð eða sleppa því

thrainn25 replied:
Já skal skipti um orð.

gigjamarin commented on code/Rokstudningur.markdown:
```diff
@@ -0,0 +1,13 @@
+**Reglulega segðin**
```
Spurning að hafa verkefnalýsinguna hérna efst þannig maður viti svona aðeins meira hvað er í gangi

thrainn25 replied:
Já skal bæta henni við.

gigjamarin commented on code/Rokstudningur.markdown:
```diff
@@ -0,0 +1,13 @@
+
+**tökuhópurinn ([^,]+) sem kemur fram fjórum sinnum**
+Passar við einn staf sem er ekki til í listanum hér að neðan [^,].
+Passar við fyrri táknið á milli eins og ótakmarkaðs tíma, eins oft og mögulegt er, gefur til baka eftir þörfum.
```
Passar við fyrra* 
ertu að tala um að + merkið? í línu 6. 
Það þarf aðeins að laga þessa línu ég skil ekki alveg.

thrainn25 replied:
Já útskýri betur

gigjamarin commented on code/Rokstudningur.markdown:
```diff
@@ -0,0 +1,13 @@
+**Meta escape**
+\s passar við hvaða hvíta staf sem er (jafngildir [\r\n\t\f\v ])
+
+**handtökuhópur (.+), aftasti hópurinn**
```
hvað er handtökuhópur? spurning að nota betra orð yfir þetta.

thrainn25 replied:
Já breyti þessu líka.

gigjamarin commented on code/regex_reorder.py:
```diff
@@ -32,12 +33,25 @@ def lesa_skra(file_path):
+                # Sameinum partana í réttri röð
+                endurröðuð_lina = '\t'.join([address, phone_number, f'{last_name}, {first_middle_names}'])
+                endurröðuðar_linur.append(endurröðuð_lina)
+
```
Þetta keyrist ekki rétt hjá mér með skipuninni python3 code/regex_reorder.py --infile data/nafn_heimilisfang_simanumer.csv --outfile data/heimilisfang_simanumer_nafn.tsv

Ég sé ekki alveg hvað vandamálið er en þú kannski skoðar það aðeins

thrainn25 replied:
Já skoða það

gigjamarin reviewed (changes requested):
Það þarf að laga orðarfar að köflum. Svo virðist forritið ekki keyrast rétt þegar ég keyri það hjá mér. Sjá frekari comment inni á skjalinu.

asdisgudmunds reviewed (commented):
Kóðinn virkar heldur ekki hja mér

asdisgudmunds reviewed (commented):
Breytti kóðanum aðeins og þetta virkar hjá mer: 

import os
import argparse
import re


def parse_arguments():
    """
    Lesa inn argument frá skipanalínu
    :return: (ArgumentParser) parser með argumentunum
    """

    parser = argparse.ArgumentParser(description="""Endurraða csv-skrá sem hefur röðunina: 
    1) eiginnafn millinafn kenninafn, 2) heimilisfang, 3) símanúmer.    
    Úttaksskráin verður endurröðuð í tsv röð:
    1) heimilisfang, 2) símanúmer, 3) kenninafn, eigninnafn millinafn.
    """)
    parser.add_argument('--infile', required=True,
                        help='Slóð að inntaksskrá með upplýsingum.')
    parser.add_argument('--outfile', required=True,
                        help='Slóð að úttaksskrá með endurröðuðum upplýsingum.')
    return parser.parse_args()


def lesa_skra(file_path):
    """
    Lesa inntaksskrá og skila lista af línum
    :param file_path: (str) Slóð að skrá
    :return:          (list) Listi af línum
    """

    with open(file_path, 'r') as file:
        return file.readlines()


def endurraða_skra(linur):
    endurröðuðar_linur = []
    for lina in linur:
        # Notum reglulega segð til þess að skipta línunum upp í parta
        pattern = r'([^,]+)\s([^,]+),\s([^,]+),\s([^,]+),\s(.+)'
        match = re.match(pattern, lina.strip())
        if match:
            ## print(len(match.groups()))
            name, last_name, address, address1, phone_number = match.groups()
            ## print(f"{name}, {last_name}, {address}, {phone_number}")

            
            # Sameinum partana í réttri röð
            endurröðuð_lina = '\t'.join([address, address1, phone_number, f'{last_name}, {name}'])
            endurröðuðar_linur.append(endurröðuð_lina)

    return endurröðuðar_linur

def skrifa_nidurstodur(output_file, linur):
    """
    Skrifar niðurstöður í úttaksskrá
    :param output_file: (str) Slóð að úttaksskrá
    :param linur:       (list) Listi af línum
    :return:            None
    """
    with open(output_file, 'w') as file:
        for lina in linur:
            file.write(lina + '\n')


def main():
    """
    Keyrslufall sem les inn skrá, endurraðar henni og skrifar niðurstöður í nýja skrá.
    :return: None
    """

    args = parse_arguments()

    # Athuga hvort skráar endingar séu réttar
    if not args.infile.endswith('.csv'):
        print('Inntaksskrá þarf að vera csv skrá.')
        return
    if not args.outfile.endswith('.tsv'):
        print('Úttaksskrá þarf að vera tsv skrá.')
        return
    # Athuga hvort inntaksskrá sé til
    if not os.path.exists(args.infile):
        print('Inntaksskrá fannst ekki.')
        return

    # Lesa línur úr skránni
    linur = lesa_skra(args.infile)

    # Endurraða hverri línu í skránni
    linur = endurraða_skra(linur)

    # Skrifa niðurstöðurnar í nýja skrá
    skrifa_nidurstodur(args.outfile, linur)


if _name_ == "_main_":
    main()

thrainn25 commented:
Okey geggjað Ásdís, skoða þetta

thrainn25 committed:
Spurning 3

Lagfræslur eftir comment

thrainn25 committed:
Lagfæringar fyrir keyrslu

Við lagfæringu fór óvart if_name_ í vesen neðst. Lagði það og við það býr kóðinn til nýtt skjal með réttri útkomu.

asdisgudmunds reviewed (approved):
Flott og virkar

gigjamarin reviewed (approved):
mjög flott. Keyrist rétt núna líka
